fix(page-data): add key to allPages (#10625)
withastro/astro: .changeset/great-turtles-greet.md

---
"astro": minor
---

> Multiple pages can now use the same component as an `entrypoint`. This change is purely internal, and aligns the build process with the behaviour in the development server.

Reviewer: ```suggestion
Adds the ability for multiple pages to use the same component as an `entrypoint` when building an Astro integration. This change is purely internal, and aligns the build process with the behaviour in the development server.
```
@goulvenclech - Another thought:

Also reading back through the old issue and comments there, perhaps it is worth calling out (if true!) that this relates exclusively to building integrations? I know when people get deep into their own features, it can be easy to have a lot of context in one's head. But, this will be read by people simply updating their own site's version of Astro, and providing some guidance as to whether or not this is something that affects them is often helpful!

(Revisiting this also gives me the opportunity to use the format I *really* want, which is to start with my present tense verb like "Adds"/"Fixes"/"Removes"... 😄 )

Author: > when building an Astro integration.

I'm not sure we want to add this... I think currently the `injectRoute` function is the only way to use this change (maybe I'm wrong) so it's not wrong.... BUT this PR removed the technical limitation in the build process, not in the integration API. So if there is another way to inject routes or there will be one in the future, everyone will be able to use this, integration or not.

> (Revisiting this also gives me the opportunity to use the format I really want, which is to start with my present tense verb like "Adds"/"Fixes"/"Removes"... 😄 )

Noted for next time!

Reviewer: @goulvenclech 

> Adds the ability for multiple pages to use the same component as an `entrypoint` when building an Astro integration.
 
Thanks for the reply and would love to get this just right! Here's my thinking, and tell me where I'm off:

This does say "when building an Astro integration" as a *use case* (not that the Integration API itself was actually changed), so if we're aiming at helping readers figure out whether or not the change affects them, I think this might be helpful even if it doesn't describe *what* internal change now allows them this new flexibility when building an integration?

I will also note that this is gonna get buried in a changelog soon, so I think it's OK to write for what this unlocks and allows *now* vs thinking about potential uses in the future. If and when it becomes possible for someone to use this in another situation, presumably we will have docs for that new situation?

Reviewer 2: I agree with Sarah regarding this being a "marginal use case". I don't see any need to add this on Astro Docs.

> Multiples pages can now have the same component as entrypoint. This is purely internal, this aligns the build process with the behaviour in development server, no public API changes.

Reviewer: ```suggestion
Multiple pages can now use the same component as an `entrypoint`. This change is purely internal, and aligns the build process with the behaviour in the development server.
```
just some tiny language edits! Please adjust if anything is inaccurate!
